test/khash_test.erl

```diff
@@ -0,0 +1,443 @@
+-module(khash_test).
+
+-compile([export_all]).
+
+-define(NODEBUG, true).
```

Comment:
Well, I use rebar 2.5.1 under Ubuntu, tried both to R14B01 and R17.6.5 and I definitely got tests compiled with TEST flag

```
1> ls().
gen_term.beam       gen_term.erl        khash.beam          
khash.erl           khash_test.beam     khash_test.erl      

ok
2> beam_lib:chunks("khash_test.beam", [compile_info]).
{ok,{khash_test,[{compile_info,[{options,[export_all,
                                          {outdir,".eunit"},
                                          {d,'TEST'},
                                          debug_info,debug_info,
                                          {i,"include"},
                                          return_errors,return_warnings,export_all]},
                                {version,"4.7.2"},
                                {time,{2015,7,14,22,29,36}},
                                {source,"/home/vagrant/git/couchdb/src/khash/test/khash_test.erl"}]}]}}
```

My runtime 5 sec which kind of feels like tests are actually running and if I screw with the assertions they are properly failing.

Can you please check compile_info on your test's beam? I mean I don't mind to remote `NODEBUG` flag, I'm not that attached to it, but just for sake of understanding of what's going on here.

[https://github.com/rebar/rebar/blob/master/src/rebar_erlc_compiler.erl#L234](This line)